internal/resources/syntheticmonitoring/resource_check_alert.go

```diff
@@ -0,0 +1,226 @@
+	AlertNameTLSTargetCertificateCloseToExpiring = "TLSTargetCertificateCloseToExpiring"
+)
+
+func resourceCheckAlerts() *common.Resource {
+	return common.NewLegacySDKResource(
+		common.CategorySyntheticMonitoring,
+		"grafana_synthetic_monitoring_check_alerts",
+		resourceCheckAlertID,
+		&schema.Resource{
+			Description: `
+Manages alerts for a check in Grafana Synthetic Monitoring.
+
+* [Official documentation](https://grafana.com/docs/grafana-cloud/synthetic-monitoring/configure-alerts/)
+* [API documentation](https://github.com/grafana/synthetic-monitoring-api-go-client/blob/main/docs/API.md#alerts)
+`,
+
+			CreateContext: withClient[schema.CreateContextFunc](resourceCheckAlertCreate),
+			ReadContext:   withClient[schema.ReadContextFunc](resourceCheckAlertRead),
+			UpdateContext: withClient[schema.UpdateContextFunc](resourceCheckAlertUpdate),
+			DeleteContext: withClient[schema.DeleteContextFunc](resourceCheckAlertDelete),
+
+			Importer: &schema.ResourceImporter{
+				StateContext: schema.ImportStatePassthroughContext,
+			},
+
+			Schema: map[string]*schema.Schema{
+				"check_id": {
+					Description: "The ID of the check to manage alerts for.",
+					Type:        schema.TypeInt,
+					Required:    true,
+					ForceNew:    true,
+				},
+				"alerts": {
+					Description: "List of alerts for the check.",
+					Type:        schema.TypeSet,
+					Required:    true,
+					ConfigMode:  schema.SchemaConfigModeAttr,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"name": {
+								Description: "Name of the alert. Must be one of: " + AlertNameProbeFailedExecutionsTooHigh + ", " + AlertNameTLSTargetCertificateCloseToExpiring,
+								Type:        schema.TypeString,
+								Required:    true,
+								ValidateFunc: validation.StringInSlice([]string{
+									AlertNameProbeFailedExecutionsTooHigh,
+									AlertNameTLSTargetCertificateCloseToExpiring,
+								}, false),
+							},
+							"threshold": {
+								Description: "Threshold value for the alert.",
+								Type:        schema.TypeFloat,
+								Required:    true,
+							},
+							"period": {
+								Description: "Period for the alert threshold. Required only for " + AlertNameProbeFailedExecutionsTooHigh + " alerts. One of: `1m`, `2m`, `5m`, `10m`, `15m`, `20m`, `30m`, `1h`.",
+								Type:        schema.TypeString,
+								Required:    false,
+								Optional:    true,
+								Default:     "",
+								ValidateFunc: validation.StringInSlice([]string{
+									"", "1m", "2m", "5m", "10m", "15m", "20m", "30m", "1h",
+								}, false),
+							},
+						},
+					},
+				},
+			},
+		},
+	)
+}
+
+func resourceCheckAlertCreate(ctx context.Context, d *schema.ResourceData, c *smapi.Client) diag.Diagnostics {
+	checkID := int64(d.Get("check_id").(int))
+
+	// Validate that the check exists
+	if err := validateCheckExists(ctx, c, checkID); err != nil {
+		return diag.FromErr(err)
+	}
```

Comment:
Do we have to do this from the client considering that the backend will already perform this validation as part of the "update alerts" request?